gax/src/main/java/com/google/api/gax/retrying/ExponentialPollAlgorithm.java

```diff
@@ -65,6 +65,6 @@ public boolean shouldRetry(TimedAttemptSettings nextAttemptSettings) throws Poll
     if (super.shouldRetry(nextAttemptSettings)) {
       return true;
     }
-    throw new PollException();
+    throw new PollException("timeout or retries exceeded");
```

Comment:
Can it be phrased like "Total timeout or maximum number of attempts exceeded"? Also, we can make it even more specific (should be useful, taking into account that most proabbly after seeing this message users will want to increase the limits and to increase them they will need to know what the current limits are). The `nextAttemptSettings.getGlobalSettings().getTotalTimeout()` gives us the actual `totalTimeout` value and `nextAttemptSettings.getGlobalSettings().getMaxAttempts()` gives the maximum attempts number (note if it is `0` it means infinite number of attempts).

Reply:
I'm not sure what kind of message you have in mind. Is it something like

```
Total timeout or maximum number of attempts exceeded. Current settings is ...
```
?

Reply:
yes, something like that would be good. Or at minimum the message should specify the total timeout was exceeded (just "timeout" is too generic), and use terms "attempts" not "retries", because in case of polling each attempt is not really a retry.